pkg/executor/importer/kv_encode.go

```diff
@@ -64,13 +69,22 @@ func NewTableKVEncoder(
 	if err != nil {
 		return nil, err
 	}
+	insertRowNeedCast := make([]bool, len(baseKVEncoder.Columns))
+	for i := 0; i < len(ti.InsertColumns); i++ {
+		insertCol := ti.InsertColumns[i].ToInfo()
+		offset := ti.InsertColumns[i].Offset
+		if offset < len(insertRowNeedCast) {
+			insertRowNeedCast[offset] = !(baseKVEncoder.Columns[offset].ToInfo().FieldType.Equal(&insertCol.FieldType))
```

Comment:
Can you add comments to explain the meaning of `baseKVEncoder`'s table and `ti`'s table? From a quick glance, I didn't find which one is the "table to be imported" and which one is the table associated with input datum.

Reply:
I'm sorry, but I don't know what the specific difference is between `ti.InsertColumns` and  `baseKVEncoder.Columns`, @D3Hunter Could you help to explain this?

Reply:
I just afraid that they are built from the same table, so here the two columns are always equal. I'll check the detailed code.

Reply:
look like `insertRowNeedCast ` is useless, and I already remove it.